CSharpMath.Editor/Extensions/MathList.cs

```diff
@@ -14,83 +17,96 @@ partial class Extensions {
-          if (currentAtom.Subscript.IsEmpty() && currentAtom.Superscript.IsEmpty())
-            throw new SubIndexTypeMismatchException(index);
-          if (atom.Subscript.IsNonEmpty() || atom.Superscript.IsNonEmpty())
-            throw new ArgumentException("Cannot fuse with an atom that already has a subscript or a superscript");
-          atom.Subscript.Append(currentAtom.Subscript);
-          atom.Superscript.Append(currentAtom.Superscript);
-          currentAtom.Subscript.Clear();
-          currentAtom.Superscript.Clear();
-          var atomIndex = index.AtomIndex;
-          // Prevent further subindexing inside BetweenBaseAndScripts
-          if (advanceType != MathListSubIndexType.None
-              && index.LevelDown() is MathListIndex levelDown) index = levelDown.Next;
-          self.InsertAtAtomIndexAndAdvance(atomIndex + 1, atom, ref index, advanceType);
-          break;
-        case MathListSubIndexType.Degree:
-        case MathListSubIndexType.Radicand:
-          if (!(self.Atoms[index.AtomIndex] is Atoms.Radical radical))
-            throw new SubIndexTypeMismatchException(typeof(Atoms.Radical), index);
-          if (index.SubIndexType == MathListSubIndexType.Degree)
-            radical.Degree.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          else radical.Radicand.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          break;
-        case MathListSubIndexType.Numerator:
-        case MathListSubIndexType.Denominator:
-          if (!(self.Atoms[index.AtomIndex] is Atoms.Fraction frac))
-            throw new SubIndexTypeMismatchException(typeof(Atoms.Fraction), index);
-          if (index.SubIndexType == MathListSubIndexType.Numerator)
-            frac.Numerator.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          else
-            frac.Denominator.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          break;
-        case MathListSubIndexType.Subscript:
-          self.Atoms[index.AtomIndex].Subscript.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          break;
-        case MathListSubIndexType.Superscript:
-          self.Atoms[index.AtomIndex].Superscript.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          break;
-        case MathListSubIndexType.Inner:
-          if (!(self.Atoms[index.AtomIndex] is Atoms.Inner inner))
-            throw new SubIndexTypeMismatchException(typeof(Atoms.Inner), index);
-          inner.InnerList.InsertAndAdvance(ref index.SubIndex, atom, advanceType);
-          break;
-        default:
-          throw new SubIndexTypeMismatchException(index);
+      switch (index.SubIndexInfo) {
+        case null:
+          return self.InsertAtAtomIndexAndAdvance(atomIndex, atom, advanceType);
+        case (MathListSubIndexType type, MathListIndex subIndex):
+          switch (type) {
+            case MathListSubIndexType.BetweenBaseAndScripts: // TODO: finish this section
```

Comment:
I don't really understand this case. The docs on `BetweenBaseAndScripts` has `The position in the subindex is an index into the nucleus, must be 1` and I don't understand that exactly. Could you help with this part @Happypig375 ?

Reply:
The AtomIndex must be 1 for SubIndexType BetweenBaseAndScripts.

Reply:
I got that far...

Reply:
In `12^3`, is there a MathListIndex of `2`?

Reply:
Yes. The right boundary of the whole MathList.

Reply:
What is the MathListIndex of `12^3`?

Reply:
Yes. The right boundary of the whole MathList. (for insertions)

Cursor position|MathListIndex
-|-
‸12^3|0
1‸2^3|1
12‸^3|1, BetweenBaseAndScripts:1
12^‸3|1, Superscript:0
12^3‸|1, Superscript:1 (to the right of 3) OR 2 (to the right of the entire MathList)